php/Blockonomics.php

```diff
@@ -753,39 +809,60 @@ public function check_callback_secret($secret){
     }
 
     public function save_transaction($value, $order, $wc_order){
-        if (!metadata_exists('post',$wc_order->get_id(),'blockonomics_'. $order['crypto'] .'_txid', $order['txid']) )  {
-            update_post_meta($wc_order->get_id(), 'blockonomics_'. $order['crypto'] .'_txid', $order['txid']);
-            update_post_meta($wc_order->get_id(), 'expected_'. $order['crypto'], $order['satoshi']/1.0e8);
+        $txid_metavalue = get_post_meta($order['order_id'], $key = 'blockonomics_payments_txid');
+        if (empty($txid_metavalue) || $txid_metavalue[0] == $order['txid']){
+            update_post_meta($wc_order->get_id(), 'blockonomics_payments_txid', $order['txid']);
+        }
+        else{
+            $all_txids_for_order = $this->get_order_txids($order['order_id']);
+            if (!strpos($txid_metavalue[0], $order['txid'])){
+                update_post_meta($wc_order->get_id(), 'blockonomics_payments_txid', $all_txids_for_order . $order['txid'] .' ('.$order['crypto'].')');
+            }
         }
     }
 
     public function update_paid_amount($status, $value, $order, $wc_order){
         $network_confirmations = get_option("blockonomics_network_confirmation",2);
-        if ($status >= $network_confirmations && !metadata_exists('post',$wc_order->get_id(),'paid_'. $order['crypto']) )  {
-          update_post_meta($wc_order->get_id(), 'paid_'. $order['crypto'], $value/1.0e8);
-          $status = $this->check_paid_amount($status, $value, $order, $wc_order);
+        if ($status >= $network_confirmations && !metadata_exists('post',$wc_order->get_id(),'_paid_to_'. $order['address']) )  {
+          update_post_meta($wc_order->get_id(), '_paid_to_'. $order['address'], $value/1.0e8);
+          $order['payment_status'] = 2;
+          $order = $this->check_paid_amount($status, $value, $order, $wc_order);
           $this->update_temp_draw_amount($value);
-          return $status;
+          return $order;
         }
-        return $status;
+        // since $status < $network_confirmations payment_status should be 1 i.e. payment in progress if payment is not already completed
+        if ($order['payment_status'] != 2){
+            $order['payment_status'] = 1;
```

Comment:
In the 2nd if statement, meta_key check is needed so that multiple callbacks don't update temp amount each time callback is hit. So, I have written it as
`if ($callback_status >= $network_confirmations && !metadata_exists('post',$wc_order->get_id(), $meta_key) ) {`
where `$meta_key = '_paid_to_'. $order['address'];`

Reply:
I do not believe this check is necessary since the payment status is also updated `$order['payment_status'] = 2;` on the 1st callback where the `$status >= $network_confirmations`. Consecutive callbacks should then return the existing order and will not update the paid amount again, due to the payment status check.
```
if ($order['payment_status'] == 2) {
    return $order;
}
```